src/Fls.Results.Test/ResultFunctionsTests.cs

```diff
@@ -53,5 +54,122 @@ public void BindTest()
 
             Assert.Equal(expectedResult, actualResult);
         }
+
+        [Fact]
+        public async void BindAsyncTest_1()
```

Comment:
Suppose one of these tests fails and the output says 'BindAsyncTest_1 failed' how do I make out what scenario isn't working. Make test names say in their names what they mean to test and don't be afraid of long names, just of the meaningless ones.